grpc-api/src/main/java/io/deephaven/grpc_api/session/SessionState.java

```diff
@@ -372,8 +423,8 @@ public static boolean isExportStateFinal(final ExportNotification.State state) {
         // used to manage liveness of dependencies (to prevent a dependency from being released before it is used)
         private List<ExportObject<?>> parents = Collections.emptyList();
 
-        // used to detect when this object is ready for export
-        volatile int dependentCount = -1;
+        // used to detect when this object is ready for export (is visible for atomic int field updater)
+        protected volatile int dependentCount = -1;
```

Comment:
What happens if you keep the field private? Feels kind of yucky to make the field visible, since it's a public class. You could always make it a `static` class with an explicit reference to the enclosing `SessionState`.

Reply:
The issue is giving proper access to DEPENDENT_COUNT_UPDATER. The UPDATER is static and on the outer class. This is an inner class that captures the outer class closure. I can make it package-private instead of protected. Making it private however gives this error:

```
grpc-api_1    | java.lang.IllegalAccessException: Class io.deephaven.grpc_api.session.SessionState can not access a member of class io.deephaven.grpc_api.session.SessionState$ExportObject with modifiers "private volatile"
```

I can't move the field UPDATER into the inner class because you can't have a static member inside of a non-static inner class.

Reply:
Can you try the "make it static" approach and see what happens?